### DISCUSSION TITLE
can get http request body value in tomcat plugin?

### DISCUSSION THREAD

**Discussion by JihanLin** [apache/skywalking#10783] [General]
Hello,
can i get http request body value in tomcat plugin?
now,just get http request param and method,header.have any way get body?
thank you very much.

**Comment by lujiajing1126** [comment]
Basically NO. The stream of the body can only be consumed once in the default implementation.

**Reply by JihanLin** [reply]
thank u

**Comment by wu-sheng** [comment]
The only way is that,  you build a plugin to intercept the stream(bytes) read by the web server, but you have to decode by your own.
But, SkyWalking community wouldn't recommend you to do so. It would be a huge performance impact to your systems but provide little values.

**Reply by JihanLin** [reply]
thank u
